flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/InputTypeStrategies.java

```diff
@@ -356,6 +357,9 @@ public static InputTypeStrategy commonArrayType(int count) {
         return new CommonArrayInputTypeStrategy(ConstantArgumentCount.of(count));
     }
 
+    public static InputTypeStrategy arrayComparableElementType() {
```

Comment:
How about e.g. `arrayFullyComparableElementType` to reflect the type of comparison? Or just instantiate the strategy inline.

Reply:
Ok, I will change it name.